homeassistant/components/cover/somfy.py

```diff
@@ -0,0 +1,95 @@
+"""
+Support for Somfy Covers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/cover.somfy/
+"""
+
+from homeassistant.components.cover import CoverDevice, ATTR_POSITION, \
+    ATTR_TILT_POSITION
+from homeassistant.components.somfy import DOMAIN, SomfyEntity, DEVICES
+
+DEPENDENCIES = ['somfy']
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Somfy cover platform."""
+    from pymfy.api.devices.category import Category
+    categories = {Category.ROLLER_SHUTTER.value, Category.INTERIOR_BLIND.value,
+                  Category.EXTERIOR_BLIND.value}
+
+    devices = hass.data[DOMAIN][DEVICES]
+    for cover in devices:
+        if categories & set(cover.categories):
+            add_entities([SomfyCover(cover, hass)])
+
+
+class SomfyCover(SomfyEntity, CoverDevice):
+    """Representation of a Somfy cover device."""
+
+    def __init__(self, device, hass):
+        """Initialize the Somfy device."""
+        from pymfy.api.devices.blind import Blind
+        super().__init__(device, hass)
+        self.cover = Blind(self.device, self.api)
+
+    def close_cover(self, **kwargs):
+        """Close the cover."""
+        self.cover.close()
+
+    def open_cover(self, **kwargs):
+        """Open the cover."""
+        self.cover.open()
+
+    def stop_cover(self, **kwargs):
+        """Stop the cover."""
+        self.cover.stop()
+
+    def set_cover_position(self, **kwargs):
+        """Move the cover shutter to a specific position."""
+        position = kwargs.get(ATTR_POSITION)
+        self.cover.set_position(100 - position)
+
+    @property
+    def current_cover_position(self):
+        """Return the current position of cover shutter."""
+        position = None
+        if self.has_capability('position'):
+            position = 100 - self.cover.get_position()
```

Comment:
Is this doing I/O?

State properties shouldn't do any I/O. We can do that in `update` and cache the value in an attribute.

Reply:
There is no I/O all the states  are retrieved from the device object updated with the method [update_all_devices](https://github.com/home-assistant/home-assistant/pull/19548/files#diff-5f6fcaa13fcf30fc2741aedd6cc2666bR178)